utils/run-topologically/profiler.js

```diff
@@ -0,0 +1,86 @@
+"use strict";
+
+const fs = require("fs-extra");
+const upath = require("upath");
+
+const hrtimeToMicroseconds = hrtime => {
+  return (hrtime[0] * 1e9 + hrtime[1]) / 1000;
+};
+
+const range = len => {
+  return Array(len)
+    .fill()
+    .map((_, idx) => idx);
+};
+
+const generateOutputname = () => {
+  const now = new Date(); // 2011-10-05T14:48:00.000Z
+  const datetime = now.toISOString().split(".")[0]; // 2011-10-05T14:48:00
+  const datetimeNormalized = datetime.replace(/-|:/g, ""); // 20111005T144800
+  return `Lerna-Profile-${datetimeNormalized}.json`;
+};
+
+class Profiler {
+  constructor({ concurrency, log, profile, profileLocation, rootPath }) {
+    this.events = [];
+    this.profile = profile;
+    this.log = log;
+    this.profileLocation = profileLocation;
+    this.rootPath = rootPath;
```

Comment:
I'm not sure why you need `profileLocation` _and_ `rootPath`, the defaulting would be wherever the config is created, not in this class's `output` method.

Reply:
I can move the logic of creating the output path out of the `output` method and place it in the `constructor`. Does that sound better?

Reply:
We could even [coerce](https://github.com/yargs/yargs/blob/master/docs/api.md#coercekey-fn) the `--profile-location` option with `upath.resolve` (proxied from `path` with internal normalization) and accept `opts.profileLocation` as-is.